src/QsCompiler/Core/Transformations/SyntaxTreeTransformation.fs

```diff
@@ -1,39 +1,36 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
 namespace Microsoft.Quantum.QsCompiler.Transformations.Core
 
-#nowarn "44" // TODO: RELEASE 2022-09, reenable after the overrides (*) are removed
+#nowarn "44" // RELEASE 2022-09: Re-enable after updating the ICommonTransformation implementations.
 
 open System.Collections.Immutable
+open Microsoft.Quantum.QsCompiler.DataTypes
 open Microsoft.Quantum.QsCompiler.SyntaxExtensions
 open Microsoft.Quantum.QsCompiler.SyntaxTree
-
+open Microsoft.Quantum.QsCompiler.Transformations.Core
 
 // setup for syntax tree transformations with internal state
 
-type SyntaxTreeTransformation<'T> private (state: 'T, options: TransformationOptions, _internal_: string) =
-    inherit CommonTransformationNodes()
-
+type SyntaxTreeTransformation<'T>(state, options) as this =
```

Comment:
In this PR I have seen a couple of instances where the type of a parameter is no longer explicitly described. I personally find it useful when reading code. Is not including the type of a parameter a general practice in F#? If so, why is it favored?

Reply:
Yes, F# has a strong culture of type inference that tends to prefer omitting type annotations when not needed. But also important is that there's no support in the tooling for requiring type annotations. As far as I know, there's no way to enable a compiler warning or a lint for functions that are missing parameter or return types. The standard F# way is to create a signature file (.fsi), which does enforce that every public or internal declaration in the file has an explicit type.